src/bsp/src/posix/src/UserInterface.cpp

```diff
@@ -0,0 +1,50 @@
+#include "UserInterface.h"
+#include "ros/ros.h"
+#include <cstdarg>
+
+int UserInterface::print(const char* format, va_list args) const {
+    int ret = vprintf(format, args);
+    printf("\n\r");
+    return ret;
+}
+
+int UserInterface::print(const char* format, ...) const {
+    va_list args;
+    va_start(args, format);
+    int retValue = print(format, args);
+    va_end(args);
+
+    return retValue;
+}
+
+int UserInterface::logInfo(const char* format, va_list args) const {
+    const int bufferSize = 512;
+    char buffer[bufferSize];
```

Comment:
@xgroleau For the buffer, it is a bug on posix for some reason. The workaround is present in HiveMind. Happily, this is not required on the target build, so we save those buffers where it really matters.

Reply:
I'm fine with a buffer on the Unix build since you need to format it and pass it to ROS anway, the comment you are referring to was on the ESP build.

I'm not sure what is the bug you are referring to though

![image](https://user-images.githubusercontent.com/31734358/104818379-3c960200-57f5-11eb-94e5-8027a34f90c6.png)